theora_image_transport/src/theora_publisher.cpp

```diff
@@ -69,81 +70,73 @@ TheoraPublisher::~TheoraPublisher()
   th_info_clear(&encoder_setup_);
 }
 
-void TheoraPublisher::advertiseImpl(ros::NodeHandle &nh, const std::string &base_topic, uint32_t queue_size,
-                                    const image_transport::SubscriberStatusCallback  &user_connect_cb,
-                                    const image_transport::SubscriberStatusCallback  &user_disconnect_cb,
-                                    const ros::VoidPtr &tracked_object, bool latch)
+void TheoraPublisher::advertiseImpl(
+  rclcpp::Node::SharedPtr node,
+  const std::string &base_topic,
+  rmw_qos_profile_t custom_qos)
 {
+  // TODO: how to port these configurations to ROS2?
   // queue_size doesn't account for the 3 header packets, so we correct (with a little extra) here.
-  queue_size += 4;
+  // queue_size += 4;
   // Latching doesn't make a lot of sense with this transport. Could try to save the last keyframe,
```

Comment:
Latching references could be cleared. Maybe there should be logic to warn that durability is not recommended if it's passed into the system?